lib/collections.js

```diff
@@ -46,9 +57,30 @@ var longAndShort = {};
 longAndShort['longest'] = '';
 longAndShort['shortest'] = paragraph;
 
-/* your code starts here */
+for (i = 0; i < normalizedWords.length; i++) {
+  var words = normalizedWords[i];
+  longAndShort[words] = words.length;
+
+} // words with keys and their length
+
+var longest = 0;
+var shortest = 0;
+var wordLength = normalizedWords;
+var k = 0;
+
+for (var j = 0; j < wordLength.length; j++) {
+if (longest < wordLength[j].length) {
+    longest = wordLength[j];
+    console.log("longest word")
+  } else {
```

Comment:
That's not right - it shouldn't be an else here. Just because we failed the first test (not longer than the longest word) that doesn't automatically mean that we pass the second test (shorter than the shortest word.)